src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -195,53 +201,84 @@ private LiKafkaProducerImpl(LiKafkaProducerConfig configs,
       K key = producerRecord.key();
       V value = producerRecord.value();
       Long timestamp = producerRecord.timestamp() == null ? System.currentTimeMillis() : producerRecord.timestamp();
-      Integer partition = producerRecord.partition();
-      Future<RecordMetadata> future = null;
-      UUID messageId = getUuid(key, value);
       if (LOG.isTraceEnabled()) {
-        LOG.trace("Sending event: [{}, {}] with key {} to kafka topic {}",
-            messageId.toString().replaceAll("-", ""),
-            value.toString(),
-            (key != null) ? key.toString() : "[none]",
+        LOG.trace("Sending record with [key={}, value={}] to kafka topic {}",
+            key == null ? "[none]" : key,
+            value == null ? "[none]" : value,
             topic);
       }
       byte[] serializedValue;
       byte[] serializedKey;
       try {
         serializedValue = _valueSerializer.serialize(topic, value);
         serializedKey = _keySerializer.serialize(topic, key);
-      } catch (Throwable t) {
+      } catch (Exception t) {
         // Audit the attempt and the failure.
         _auditor.record(topic, key, value, timestamp, 1L, 0L, AuditType.ATTEMPT);
         _auditor.record(topic, key, value, timestamp, 1L, 0L, AuditType.FAILURE);
         throw new KafkaException(t);
       }
-      int sizeInBytes = (serializedKey == null ? 0 : serializedKey.length)
-          + (serializedValue == null ? 0 : serializedValue.length);
-      // Audit the attempt.
-      _auditor.record(topic, key, value, timestamp, 1L, (long) sizeInBytes, AuditType.ATTEMPT);
-      // We wrap the user callback for error logging and auditing purpose.
-      Callback errorLoggingCallback =
-          new ErrorLoggingCallback<>(messageId, key, value, topic, timestamp, sizeInBytes, _auditor, callback);
-      if (_largeMessageEnabled && serializedValue != null && serializedValue.length > _maxMessageSegmentSize) {
-        List<ProducerRecord<byte[], byte[]>> segmentRecords =
-            _messageSplitter.split(topic, partition, timestamp, messageId, serializedKey, serializedValue);
-        Callback largeMessageCallback = new LargeMessageCallback(segmentRecords.size(), errorLoggingCallback);
-        for (ProducerRecord<byte[], byte[]> segmentRecord : segmentRecords) {
-          future = _producer.send(segmentRecord, largeMessageCallback);
-        }
+
+      Future<RecordMetadata> future = null;
+      int sizeInBytes = (serializedKey == null ? 0 : serializedKey.length) + (serializedValue == null ? 0 : serializedValue.length);
+      boolean useLargeMessageProcessing = _largeMessageEnabled && serializedValue != null && serializedValue.length > _maxMessageSegmentSize;
+      if ((!(producerRecord instanceof ExtensibleProducerRecord) && !useLargeMessageProcessing) ||
```

Comment:
I agree this is confusing.  When I get into work I will send you the truth table for this.